pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaMPI.hpp

```diff
@@ -417,13 +442,12 @@ class StateVectorCudaMPI final
      * @param wires Wires to apply gate to.
      * @param adjoint Indicate whether inverse should be taken.
      */
-    void applyMatrix(const std::complex<PrecisionT> *gate_matrix,
```

Comment:
Could we revert `ComplexT` back to `std::complex<PrecisionT>` ? We target to unify the definition of `ComplexT` of all backends. In the `LGPU` case, we plan to set `ComplexT` as `cuComplex`.

Reply:
That's what I had in mind. I don't think we want to unify `ComplexT` across backends, but the complex type will be unified in `LGPU` (a `LKokkos` use `ComplexT = Kokkos::complex`). So when `ComplexT` and `CFP_t` become the same these functions will already be rid of `std::complex<PrecisionT>`. I can still revert in the meantime if you would like.

Reply:
Rather than unify all to a single type, ComplexT should be device dependent as a templated param.

Reply:
correct, not unify the definition for backends but just want to let LGPU define ComplexT the way that LK and LQ does. ComplexT should be cuComplex type in this case. So I would like to keep the current implementation.

Reply:
To be more clear, just want to keep the std::complex<T> to make it easier to identify the changes that needed for the set complexT as cuComplex.

Reply:
Ok, will do those changes in a follow-up PR as per our discussion.